### PR TITLE
DOCK-2207: Clarify requirements for .dockstore.yml

### PR DESCRIPTION
**Pull request by svonworl**
Modifies the documentation to:
1. Note restrictions on characters in workflow/tool names.
2. Specify absolute paths for descriptor and test file paths. 
3. Add "absolute path", "primary descriptor file", and "secondary descriptor file" entries to the Dictionary.

I consulted with Ash, and we agreed that, perhaps, there wasn't enough confusion about the workflow file extension to merit documentation enhancements.

There are some rumblings in the ticket comments about whether to specify that a test parameter file is a hard requirement.  IMHO, it's sort of a gray area - we ask for a test parameter file, but it can be functionally empty - so I left the docs as-is with respect to that...

https://ucsc-cgl.atlassian.net/browse/DOCK-2207

### CODE REVIEW

File: docs/_attic/glossary_entries.py [hunk 1]
```diff
@@ -22,6 +22,12 @@
 AGC = GlossEntry("AGC",
 	acronym_full="[Amazon Genomics CLI]")
 
+AbsolutePath = GlossEntry("absolute path",
+	definition="A path that starts with the character ``/`` and contains the full set of directories necessary to resolve a file, starting from the root directory. For example: ``/Dockstore.cwl`` or ``/bin/sh``",
```
Comment (coverbeck) [inline docs/_attic/glossary_entries.py:26]: > from the root directory

I'm not sure how to work it in, and especially here where it's a generic definition, but the absolute path in the context of the bug is "relative" :) to the repository's root, not the (local) file system root.

**Review by coverbeck — APPROVED** [review]

**Review by coverbeck — APPROVED** [review]

File: docs/_attic/glossary_entries.py [hunk 2]
```diff
@@ -475,6 +481,20 @@
 	pronunciation='', 
 	seealso="[spot instance]")
 
+PrimaryDescriptorFile = GlossEntry("primary descriptor file",
+	acronym_full="",
+	definition="The main [descriptor file] that describes a workflow or tool.",
```
Comment (denis-yuen) [inline docs/_attic/glossary_entries.py:486]: main and primary seem like synonyms that aren't really shedding much light here

Comment (denis-yuen) [reply docs/_attic/glossary_entries.py:486]: maybe something like "the initial descriptor file that Dockstore starts from to find other descriptors"

**Comment by coverbeck** [comment]
@svonworl FYI, retargeted the PR to release/1.13.0.

**Review by denis-yuen — APPROVED** [review]

**Review by aofarrel — COMMENTED** [review]
Couple notes...
* The naming workflows page will not show up on the sidebar nor the homepage. If that's intended, no worries.
* Consider committing the generated files, because if you don't, someone else probably will do so on another branch and then things will get confusing.

**Comment by svonworl** [comment]
> Couple notes...
> 
> * The naming workflows page will not show up on the sidebar nor the homepage. If that's intended, no worries.
> * Consider committing the generated files, because if you don't, someone else probably will do so on another branch and then things will get confusing.

* "Naming workflows" has the same visibility as before, which is intended, I assume.
* Thanks for the heads up regarding the generated files.

**Review by aofarrel — APPROVED** [review]
Looks great, thanks!

**Pull request merged**